File: server/router/community/similar.go
package community

import (
	"github.com/gin-gonic/gin"
	v1 "github.com/wz-project/wz-backend-web/server/api/v1"
	"github.com/wz-project/wz-backend-web/server/middleware"
)

type SimilarRouter struct{}

// InitSimilarRouter 初始化相同圈子路由
func (s *SimilarRouter) InitSimilarRouter(Router *gin.RouterGroup) {
	similarRouter := Router.Group("similar").Use(middleware.OperationRecord())
	similarRouterWithoutRecord := Router.Group("similar")
	similarPublicRouter := Router.Group("similar")
	
	var similarApi = v1.ApiGroupApp.CommunityApiGroup.SimilarApi
	{
		// 需要认证和记录操作的路由
		similarRouter.POST("application", similarApi.CreateSimilarApplication)            // 创建相同圈子申请
		similarRouter.PUT("application", similarApi.UpdateSimilarApplication)             // 更新相同圈子申请
		similarRouter.DELETE("application", similarApi.DeleteSimilarApplication)          // 删除相同圈子申请
		similarRouter.PATCH("application/status", similarApi.UpdateSimilarApplicationStatus) // 更新相同圈子申请状态
	}
	{
		// 需要认证但不记录操作的路由
		similarRouterWithoutRecord.GET("application", similarApi.GetSimilarApplication)          // 获取单个相同圈子申请
		similarRouterWithoutRecord.GET("application/list", similarApi.GetSimilarApplicationList) // 获取相同圈子申请列表
		similarRouterWithoutRecord.GET("circles", similarApi.GetSimilarCircles)                  // 获取相同圈子列表
		similarRouterWithoutRecord.GET("circle/members", similarApi.GetSimilarCircleMembers)     // 获取相同圈子成员列表
	}
	{
		// 公开路由，不需要认证
		similarPublicRouter.GET("categories", similarApi.GetSimilarCategories) // 获取所有相同圈子分类
	}
}

File: web/src/api/community/similar.js
import service from '@/utils/request'

// 相同圈子申请管理API

// 获取相同圈子申请列表
export const getSimilarApplicationList = (data) => {
  return service({
    url: '/community/similar/application/list',
    method: 'get',
    params: data
  })
}

// 获取单个相同圈子申请详情
export const getSimilarApplication = (params) => {
  return service({
    url: '/community/similar/application',
    method: 'get',
    params
  })
}

// 创建相同圈子申请
export const createSimilarApplication = (data) => {
  return service({
    url: '/community/similar/application',
    method: 'post',
    data
  })
}

// 更新相同圈子申请
export const updateSimilarApplication = (data) => {
  return service({
    url: '/community/similar/application',
    method: 'put',
    data
  })
}

// 删除相同圈子申请
export const deleteSimilarApplication = (data) => {
  return service({
    url: '/community/similar/application',
    method: 'delete',
    data
  })
}

// 更新相同圈子申请状态
export const updateSimilarApplicationStatus = (data) => {
  return service({
    url: '/community/similar/application/status',
    method: 'patch',
    data
  })
}

// 相同圈子分类API

// 获取所有相同圈子分类
export const getSimilarCategories = () => {
  return service({
    url: '/community/similar/categories',
    method: 'get'
  })
}

// 相同圈子管理API

// 获取相同圈子列表
export const getSimilarCircles = (params) => {
  return service({
    url: '/community/similar/circles',
    method: 'get',
    params
  })
}

// 更新相同圈子状态
export const updateSimilarCircleStatus = (data) => {
  return service({
    url: '/community/similar/circle/status',
    method: 'patch',
    data
  })
}

// 相同圈子成员管理API

// 获取相同圈子成员列表
export const getSimilarCircleMembers = (params) => {
  return service({
    url: '/community/similar/circle/members',
    method: 'get',
    params
  })
}

// 更新成员角色
export const updateMemberRole = (data) => {
  return service({
    url: '/community/similar/circle/member/role',
    method: 'patch',
    data
  })
}

// 更新成员状态
export const updateMemberStatus = (data) => {
  return service({
    url: '/community/similar/circle/member/status',
    method: 'patch',
    data
  })
}

// 移除成员
export const removeMember = (data) => {
  return service({
    url: '/community/similar/circle/member',
    method: 'delete',
    data
  })
}
